bokeh/core/validation/check.py

```diff
@@ -29,15 +29,56 @@
 #-----------------------------------------------------------------------------
 # Globals and constants
 #-----------------------------------------------------------------------------
+__silencers__ = set()
 
 __all__ = (
     'check_integrity',
+    'silence'
 )
 
 #-----------------------------------------------------------------------------
 # General API
 #-----------------------------------------------------------------------------
 
+def silence(warning, silence=True):
+    ''' Silence a particular warning on all Bokeh models.
+
+    Args:
+        warning (Warning) : Bokeh warning to silence
+        silence (bool) : Whether or not to silence the warning
+
+    Returns:
+        A set containing the all silenced warnings
+
+    This function adds or removes warnings from a set of silencers which
+    is refered to when running ``check_integrity``. If a warning
+    is added to the silencers - then it will never be raised.
+
+    .. code-block:: python
+
+        >>> from bokeh.core.validation.warnings import EMPTY_LAYOUT
+        >>> bokeh.core.validation.silence(EMPTY_LAYOUT, True)
+        {1002}
+
+    To turn a warning back on use the same method but with the silence
+    argument set to false
+
+    .. code-block:: python
+
+        >>> bokeh.core.validation.silence(EMPTY_LAYOUT, False)
+        set()
+
+    '''
+    if not isinstance(warning, int):
+        raise ValueError('Input to silence should be a warning object '
+                         '- not of type {}'.format(type(warning)))
+    if silence:
+        __silencers__.add(warning)
+    else:
```

Comment:
```suggestion
    elif warning in __silencers__:
```

Reply:
some day if we wanted to to be fancy we could collect all the defined error and warning values, then we could raise an error of the number is not a valid error or warning. But that can happen some other time.